src/_pytest/logging.py

```diff
@@ -641,8 +647,8 @@ def _runtest_for_main(
 
             if not hasattr(item, "catch_log_handlers"):
                 item.catch_log_handlers = {}  # type: ignore[attr-defined]  # noqa: F821
-            item.catch_log_handlers[when] = log_handler  # type: ignore[attr-defined]  # noqa: F821
-            item.catch_log_handler = log_handler  # type: ignore[attr-defined]  # noqa: F821
+            item.catch_log_handlers[when] = caplog_handler  # type: ignore[attr-defined]  # noqa: F821
+            item.catch_log_handler = caplog_handler  # type: ignore[attr-defined]  # noqa: F821
```

Comment:
At some point we should also rename the attribute of the `item` from `catch_log_handler` (stems from the old name of the logging plugin, which is `pytest-catchlog`)  to `caplog_handler`

Reply:
while at it we should use the new store facility to ensure its no longer a public attribute by default

Reply:
> use the new store facility to ensure its no longer a public attribute by default

Would that make it a breaking change?

Reply:
Do we want to include that here or wait for #7162 since that will also be a breaking change?

Reply:
It is fine to include this here now, I think. `master` already contains a bunch of new features and breaking changes, so only `6.0.0` can be release out of it anyway. 👍 

We just need to add a new `7133.breaking.rst` entry:

```
The internal `logging` plugin no longer sets the `catch_log_handlers` and `catch_log_handler` attributes in test items. Those attributes are now considered private.
```